src/libraries/Common/src/System/Reflection/AssemblyNameParser.cs

```diff
@@ -0,0 +1,504 @@
+                    {
+                        flags |= (AssemblyNameFlags)(((int)AssemblyContentType.WindowsRuntime) << 9);
+                    }
+                    else
+                    {
+                        return false;
+                    }
+                }
+                else
+                {
+                    // Desktop compat: If we got here, the attribute name is unknown to us. Ignore it.
+                }
+
+                if (!TryGetNextToken(out _, out token))
+                {
+                    return false;
+                }
+            }
+
+            result = new AssemblyNameParts(name, version, cultureName, flags, pkt);
+            return true;
+        }
+
+        private static bool IsAttribute(string candidate, string attributeKind)
+            => candidate.Equals(attributeKind, StringComparison.OrdinalIgnoreCase);
+
+        private static bool TryParseVersion(string attributeValue, ref Version? version)
+        {
+#if NET8_0_OR_GREATER
+            ReadOnlySpan<char> attributeValueSpan = attributeValue;
+            Span<Range> parts = stackalloc Range[5];
+            parts = parts.Slice(0, attributeValueSpan.Split(parts, '.'));
+#else
+            string[] parts = attributeValue.Split('.');
+#endif
+            if (parts.Length is < 2 or > 4)
+            {
+                return false;
+            }
+
+            Span<ushort> versionNumbers = stackalloc ushort[4] { ushort.MaxValue, ushort.MaxValue, ushort.MaxValue, ushort.MaxValue };
+            for (int i = 0; i < parts.Length; i++)
+            {
+                if (!ushort.TryParse(
+#if NET8_0_OR_GREATER
+                    attributeValueSpan[parts[i]],
+#else
+                    parts[i],
+#endif
+                    NumberStyles.None, NumberFormatInfo.InvariantInfo, out versionNumbers[i]))
+                {
+                    return false;
+                }
+            }
+
+            if (versionNumbers[0] == ushort.MaxValue ||
+                versionNumbers[1] == ushort.MaxValue)
+            {
+                return false;
+            }
+
+            version =
+                versionNumbers[2] == ushort.MaxValue ? new Version(versionNumbers[0], versionNumbers[1]) :
+                versionNumbers[3] == ushort.MaxValue ? new Version(versionNumbers[0], versionNumbers[1], versionNumbers[2]) :
+                new Version(versionNumbers[0], versionNumbers[1], versionNumbers[2], versionNumbers[3]);
+
+            return true;
+        }
+
+        private static bool TryParseCulture(string attributeValue, out string? result)
+        {
+            if (attributeValue.Equals("Neutral", StringComparison.OrdinalIgnoreCase))
+            {
+                result = "";
+                return true;
+            }
+
+            result = attributeValue;
+            return true;
+        }
+
+        private static bool TryParsePKT(string attributeValue, bool isToken, ref byte[]? result)
```

Comment:
Why is it `ref` and not `out` when the input value is never used?

Reply:
It was me being lazy and avoiding the need of setting the result to `null` when `false` is being returned. I am going to change that to `out`.